### PR TITLE
Disable only specific source when deploying ODF

### PR DESCRIPTION
**Pull request by petr-balogh**
Fixes: #4987

Signed-off-by: Petr Balogh <pbalogh@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/resources/catalog_source.py [hunk 1]
```diff
@@ -144,3 +144,39 @@ def enable_default_sources():
...
+        source_name: Source name (e.g. redhat-operators)
+
+    """
+    logger.info(f"Disabling default source: {source_name}")
+    run_cmd(
+        constants.PATCH_SPECIFIC_SOURCES_CMD.format(
+            disable="true", source_name=source_name
+        )
+    )
+    logger.info(f"Waiting 20 seconds after disabling {source_name} sources")
```
Comment (dahorak) [inline ocs_ci/ocs/resources/catalog_source.py:163]:
```suggestion
    logger.info(f"Waiting 20 seconds after disabling {source_name} source")
```

File: ocs_ci/ocs/resources/catalog_source.py [hunk 2]
```diff
@@ -144,3 +144,39 @@ def enable_default_sources():
...
+        source_name: Source name (e.g. redhat-operators)
+
+    """
+    logger.info(f"Enabling default source: {source_name}")
+    run_cmd(
+        constants.PATCH_SPECIFIC_SOURCES_CMD.format(
+            disable="false", source_name=source_name
+        )
+    )
+    logger.info(f"Waiting 20 seconds after enabling {source_name} sources")
```
Comment (dahorak) [inline ocs_ci/ocs/resources/catalog_source.py:181]:
```suggestion
    logger.info(f"Waiting 20 seconds after enabling {source_name} source")
```

File: ocs_ci/ocs/resources/catalog_source.py [hunk 3]
```diff
@@ -144,3 +144,39 @@ def enable_default_sources():
...
+    logger.info(f"Waiting 20 seconds after disabling {source_name} sources")
+    sleep(20)
+
+
+def enable_specific_source(source_name):
+    """
+    Enable specific default source
+
+    Args:
+        source_name: Source name (e.g. redhat-operators)
```
Comment (vavuthu) [inline ocs_ci/ocs/resources/catalog_source.py:172]:
```suggestion
        source_name (str): Source name (e.g. redhat-operators)
```

File: ocs_ci/ocs/resources/catalog_source.py [hunk 4]
```diff
@@ -144,3 +144,39 @@ def enable_default_sources():
...
     logger.info("Waiting 20 seconds after enabling default sources")
     sleep(20)
+
+
+def disable_specific_source(source_name):
+    """
+    Disable specific default source
+
+    Args:
+        source_name: Source name (e.g. redhat-operators)
```
Comment (vavuthu) [inline ocs_ci/ocs/resources/catalog_source.py:154]:
```suggestion
        source_name (str): Source name (e.g. redhat-operators)
```

File: ocs_ci/ocs/resources/catalog_source.py (hunk 2 above)
Comment (petr-balogh) [reply ocs_ci/ocs/resources/catalog_source.py]: done

File: ocs_ci/ocs/resources/catalog_source.py (hunk 3 above)
Comment (petr-balogh) [reply ocs_ci/ocs/resources/catalog_source.py]: done

File: ocs_ci/ocs/resources/catalog_source.py (hunk 1 above)
Comment (petr-balogh) [reply ocs_ci/ocs/resources/catalog_source.py]: done

File: ocs_ci/ocs/resources/catalog_source.py (hunk 4 above)
Comment (petr-balogh) [reply ocs_ci/ocs/resources/catalog_source.py]: done

**Pull request merged**